build/documentation/generate-typedocs.js

```diff
@@ -0,0 +1,201 @@
+ * if an invalid third parameter is entered, the application will exit
+ */
+process.argv.forEach(function (val, index) {
+
+    var validArg = true;
+
+    if (index === 2) {
+        val === "--dry-run" ? dryRun = true : validArg = false;
+    }
+
+    if (!validArg) {
+        console.log(chalk.red(`Invalid argument used. To perform a dry-run use --dry-run`));
+        process.exit(1);
+    }
+
+});
+
+/**
+ * Run the copy readme script, then this one if successful
+ * To prevent new links constantly being appended to the readme every 
+ * time the script is run, the copy readme script is run to regenerate 
+ * files to their original form with the needed header
+ */
+
+var proc = dryRun ? exec(`node ${copyReadmeScriptPath} --dry-run`) : exec(`node ${copyReadmeScriptPath}`);
+
+proc.stdout.on('data', function(data) {
+    process.stdout.write(data);
+});
+
+proc.on('error', (err) => {
+    console.log(chalk.red(err));
+});
+
+proc.on('close', function(code) {
+    console.log(chalk.green(`${copyReadmeScript} ran successfully.`));
+    execute();
+});
+
+/**
+ * Generate TypeDocs for each package
+ */
+function execute() {
+
+    if (dryRun) {
+        console.log(`In ${destDir}, this script would...`);
+    } else {
+        console.log(`Generating API documentation using TypeDoc...`);
+    }
+
+    const packages = path.resolve(rootDir, srcDir);
+
+    glob(packages, {realpath:true}, function(error, srcFiles) {
+       
+        srcFiles.forEach((srcPath) => {    
+            
+            var valid = true;
+
+            var packageName = srcPath
+                .split(path.sep)
+                .pop();
+
+            excludedPackages.forEach((excludedPackageName) => {
+                if (packageName === excludedPackageName) {
+                    console.log(chalk.yellow(`...skip generating Typedoc API files for ${packageName}`));
+                    valid = false;
+                }
+            });
+
+            if(valid) {
+                dryRun ? console.log(`...generate Typedoc API files for ${packageName}`) : createAPIFiles(packageName, srcPath);
+                dryRun ? console.log(`...add API link to ${packageName}`) : addAPILinkToReadme(packageName);
+            }
+
+        });
+
+    });
+}
+
+/**
+ * Uses Typedoc process to generate API docs for a given package
+ * Stores them in the docs/en/packages folder 
+ */
+function createAPIFiles(packageName, srcPath) {
+    
+    var config = path.join(srcPath, '/tsconfig.json');
+    var output = path.join(destDir, packageName, 'api');
+    var file = (packageName === "fast-animation") ? path.join(srcPath, '/lib/index.ts') : path.join(srcPath, '/src/index.ts');
+
+    var typedocCmd = os.platform() == 'win32' ? 'typedoc.cmd' : 'typedoc';
+    var typedocPath = path.join('node_modules', '.bin', typedocCmd);
+
+    const typedoc = spawn(typedocPath, // the local typedoc installation is needed for the markdown theme to work
+        [
+            '--tsconfig', config,
```

Comment:
Can you add the `--excludeNotExported` flag also?

Reply:
Alright, I've added it

Reply:
thanks!